### ISSUE TITLE
Make iterating over a vector give an error

### ISSUE THREAD

**Issue by Error7Studios** [godotengine/godot-proposals#2085]
### Describe the project you are working on
(Applies to any project)

### Describe the problem or limitation you are having in your project
I had this vector:
`vec := Vector2(2, 3)`

I was trying to do this:
```gdscript
	for y in vec.y:
		for x in vec.x:
			print(Vector2(x, y))
```

But accidentally did this:
```gdscript
	for y in vec:
		for x in vec:
			print(Vector2(x, y))
```

This prints:
```
(2, 2)
```
Took me a while to realize my mistake.
Apparently `for i in <Vector2/3>` will run once and returns the x-coordinate.
This shouldn't work at all.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation
Make `for i in <Vector2/3>` give an error.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams
Give a similar error as when you have something like this:
```gdscript
	for i in GDScript:
		print("WHAT?!")
```
`Unable to iterate on object of type 'Object'.`

### If this enhancement will not be used often, can it be worked around with a few lines of script?
No

### Is there a reason why this should be core and not an add-on in the asset library?
This is built-in keyword behavior

**Comment by SIsilicon** [comment]
My guess to why this works is because the elements of vector2/3 can be accessed like an array.
`vector[0] == vector.x # etc...`

**Comment by Error7Studios** [comment]
Yeah, but even so, it doesn't behave the same as if the vector were actually replaced with an array.
```gdscript
		var array := [2, 3]

		for y in array:
			for x in array:
				print(Vector2(x, y))
```

prints:
```
(2, 2)
(3, 2)
(2, 3)
(3, 3)
```

What it currently does is just a bug waiting to happen.

**Comment by SIsilicon** [comment]
Yeah I know. Personally I'd rather if the vector acts _more_ like an array when attempting to iterate over it.

**Comment by aaronfranke** [comment]
The reason for this behavior is because Godot currently uses `Vector2` and `Vector3` for `range` under the hood.

These two for loops are identical:

```gdscript
	for i in Vector2(1, 5):
		print(i) # Prints 1 to 4
	for i in range(1, 5):
		print(i) # Prints 1 to 4
```

These two for loops are identical:

```gdscript
	for i in Vector3(1, 10, 2):
		print(i) # Prints 1, 3, 5, 7, 9
	for i in range(1, 10, 2):
		print(i) # Prints 1, 3, 5, 7, 9
```

Changing this will be a big headache, and would take away the ability for users to store a loop's start/end/step in one structure for later usage (admittedly a very rare use case).

I think this should be implemented as a GDScript warning rather than an error. What do you think @vnen?

**Comment by Error7Studios** [comment]
I see! Thanks for the explanation!
